cmd/demo/node_setup.go

```diff
@@ -109,63 +110,137 @@ func (n *node) setup() error {
 	return nil
 }
 
-// setupContracts reads from the config file whether the node should deploy or use
-// existing contract addresses.
-func (n *node) setupContracts() (err error) {
-	var adjAddr, assAddr common.Address
+func newTransactionContext() (context.Context, context.CancelFunc) {
+	return context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+}
 
-	if config.Chain.Adjudicator == "deploy" {
-		adjAddr, err = deployAdjudicator(n.cb)
-		if err != nil {
-			return
-		}
-		fmt.Println("💭 Adjudicator contract deployed")
-	} else {
-		tmpAdj, err := strToAddress(config.Chain.Adjudicator)
-		if err != nil {
-			return err
-		}
-		adjAddr = ewallet.AsEthAddr(tmpAdj)
+func parseAndValidateAdjudicator(cb echannel.ContractBackend) (common.Address, error) {
```

Comment:
```suggestion
func validateAdjudicator(cb echannel.ContractBackend) (common.Address, error) {
```
If you do the suggestion below, we can change this to single-responsibility principle.

PS: The function becomes so short without the parsing that we probably do not need it anymore.

Reply:
done